src/evict/evict_lru.c

```diff
@@ -1575,6 +1586,26 @@ __wt_cache_eviction_worker(WT_SESSION_IMPL *session, bool busy, u_int pct_full)
 	/* NOTREACHED */
 }
 
+/*
+ * __wt_evict_priority_set --
+ *	Set a tree's eviction priority.
+ */
+void
+__wt_evict_priority_set(WT_SESSION_IMPL *session, uint64_t v)
+{
+	S2BT(session)->evict_priority = v;
+}
+
+/*
+ * __wt_evict_priority_clear --
+ *	Clear a tree's eviction priority.
+ */
+void
+__wt_evict_priority_clear(WT_SESSION_IMPL *session)
+{
```

Comment:
The only "priority" we have at the moment is `(1<<20)`, set for bloom filters and internal pages.

I agree we could use something better, but that may be outside of the scope of this change; all this change does is re-implement what we have, with slightly better information hiding, and resetting the eviction priority to 0 for dead trees. (That's necessary because a dead tree with a non-zero eviction priority won't be considered for eviction unless we get "aggressive.)